### PR TITLE
feat: compound vaults onchain indexer, + chainVaults service

### PR DESCRIPTION
**Pull request by NamelsKing**

### CODE REVIEW

File: src/indexers/indexer.utils.ts [hunk 1]
```diff
@@ -63,6 +74,106 @@ export async function vaultToSnapshot(chain: Chain, vaultDefinition: VaultDefini
...
+  chain: Chain,
+  sdk: BadgerSDK,
+  vault: RegistryVault,
+): Promise<Nullable<VaultCompoundModel>> {
+  const { address, totalSupply, balance, pricePerFullShare, available } = vault;
+
+  let vaultAddr;
+
+  try {
+    vaultAddr = ethers.utils.getAddress(vault.address).toLowerCase();
```
Comment (axejintao) [inline src/indexers/indexer.utils.ts:87]: let's keep all entries as checksum consistent with rest of api

File: src/indexers/vault-compound-indexer.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const mapper = getDataMapper();
+    const query = mapper.query(
+      VaultCompoundModel,
+      { chain: chain.network },
+      { indexName: 'IndexVaultCompoundDataChain' },
+    );
+
+    try {
+      for await (const compoundVault of query) {
+        if (!prdVaultsAddrs.includes(compoundVault.address.toLowerCase())) {
```
Comment (axejintao) [inline src/indexers/vault-compound-indexer.ts:36]: let's keep all vaults supported from `devGovernance` and `strategistGuild`

Comment (axejintao) [reply src/indexers/vault-compound-indexer.ts:36]: thsi can be read from the registry

File: src/indexers/vault-compound-indexer.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      console.error(`Registry is not defined for ${chain.name}`);
+    }
+
+    if (registryVaults.length === 0) continue;
+
+    // update vaults from chain
+    await Promise.all(registryVaults.map(async (vault) => compoundVaultData(chain, sdk, vault)));
+
+    // update isProduction status, for vaults that were already saved in ddb
+    const prdVaultsAddrs = registryVaults.map((v) => v.address.toLowerCase());
```
Comment (axejintao) [inline src/indexers/vault-compound-indexer.ts:25]: checksum addr plz

File: src/indexers/vault-compound-indexer.ts (hunk 2 above)
Comment (NamelsKing) [reply src/indexers/vault-compound-indexer.ts:36]: basically, we are utilizing `sdk.vaults.loadVaults` which uses `getProductionVaults` and this is `productionVaults` in registry contract https://github.com/jack-the-pug/badger-registry/blob/2.0/contracts/BadgerRegistry.sol#L51, so we will have all prod vaults, we just need this marker to understand, if vault still presence in this map

File: src/indexers/indexer.utils.ts (hunk 1 above)
Comment (NamelsKing) [reply src/indexers/indexer.utils.ts:87]: gotcha, fixed

File: src/indexers/vault-compound-indexer.ts (hunk 3 above)
Comment (NamelsKing) [reply src/indexers/vault-compound-indexer.ts:25]: fixed

**Review by axejintao — APPROVED** [review]

**Pull request merged**